### PR TITLE
Made some enhancements to the Bundles widget

### LINKED ISSUE #3270: [Feature request] Add shortcut support to Bundles
**Is your feature request related to a problem? Please describe.**
I cannot add Microsoft Store apps to the bundle. I tried making shortcuts and use them to add them to a bundle, but it didn't worked.

**Describe the solution you'd like**
Ability to add files and folders to bundles from shortcuts. (As well something with adding MS Store apps)

### LINKED ISSUE #3449: Warning about existed bundle name
**Feature**: Attempting to create bundle with existed name will has no effect but showing something like a warning can be useful and conveniently. For example, when user types existed bundle name warning text (yellow, of course) appears under the input area.

### LINKED ISSUE #3724: Autofocus on text field
When renaming a folder and similar, please make the window jump to and autofocus to the text field so you can begin typing instead of having to click the field.

### PR DESCRIPTION
**Pull request by d2dyno1**
This PR polishes and fine-tunes [Bundles PR](https://github.com/files-community/Files/pull/2881). This is the final part that completes [Bundles](https://github.com/files-community/Files/pull/2881)

**Things yet to do**
- [x] Option to rearrange Bundle items by dragging
- [x] Select items upon `OpenPath` navigation
- [x] Ability to rearrange Bundles
- [x] Support for adding shourtcut files #3270
- [x] Display warning when creating a Bundle if the name already exists #3449
- [x] Add option to open folders in bundle in new pane
- [x] Fix memory leaks
- [x] Improve DynamicDialog (auto focus on DisplayControl) #3724 

Closes #3449
Closes #3270
Closes #3724

### CODE REVIEW

**Comment by R3voA3** [comment]
Would it be in the scope of this PR to add context menu entries to add an item to a bundle?

**Comment by crashmit** [comment]
Suggestion to improve bundles further: 
image
Currently there is no option to open bundle items in a new pane, if I'm not mistaken. Have you thought about implementing this so far?

**Comment by d2dyno1** [comment]
> Would it be in the scope of this PR to add context menu entries to add an item to a bundle?

Hmm, this is a bit of work and actually exceedes the scope of this PR. Drag & drop already works with bundles so there's no point for now to add context menu option

**Comment by d2dyno1** [comment]
> Suggestion to improve bundles further:
> image
> Currently there is no option to open bundle items in a new pane, if I'm not mistaken. Have you thought about implementing this so far?

Sounds like a good idea. I'll work on that

**Comment by R3voA3** [comment]
> > Would it be in the scope of this PR to add context menu entries to add an item to a bundle?
> 
> Hmm, this is a bit of work and actually exceedes the scope of this PR. Drag & drop already works with bundles so there's no point for now to add context menu option

Fair enough, although Drag & Drop doesn't work reliably.

File: Files/UserControls/SidebarControl.xaml [hunk 1]
```diff
@@ -160,7 +160,7 @@
                                 x:Uid="SideBarOpenInNewPane"
                                 x:Load="{x:Bind CanOpenInNewPane, Mode=OneWay}"
                                 Click="OpenInNewPane_Click"
-                                Text="Open in new pane">
+                                Text="Open in secondary pane">
```
Comment (yaichenbaum) [inline Files/UserControls/SidebarControl.xaml:163]: Can you revert this change?

File: Files/UserControls/Widgets/Bundles.xaml [hunk 2]
```diff
@@ -318,6 +324,15 @@
                                                                 <FontIcon FontFamily="{StaticResource CustomGlyph}" Glyph="&#xF106;" />
                                                             </MenuFlyoutItem.Icon>
                                                         </MenuFlyoutItem>
+                                                        <MenuFlyoutItem
+                                                            x:Uid="SideBarOpenInNewPane"
```
Comment (yaichenbaum) [inline Files/UserControls/Widgets/Bundles.xaml:328]:
```suggestion
                                                            x:Uid="BundlesOpenInNewPane"
```
Can you add a new string `BundlesOpenInNewPane`?

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

File: Files/UserControls/Widgets/Bundles.xaml (hunk 2 above)
Comment (d2dyno1) [reply Files/UserControls/Widgets/Bundles.xaml]: @yair100 all done 👍

**Review by yaichenbaum — DISMISSED** [review]
Nice work! LGTM

File: Files/Views/YourHome.xaml.cs [hunk 3]
```diff
@@ -165,5 +165,16 @@ protected override async void OnNavigatedTo(NavigationEventArgs eventArgs)
             };
             AppInstance.NavigationToolbar.PathComponents.Add(item);
         }
+
+        #region IDisposable
+
+        // TODO: This Dispose() is never called, please implement the functionality to call this function.
+        //       This IDisposable.Dispose() needs to be called to unhook events in BundlesWidget to avoid memory leaks.
```
Comment (d2dyno1) [inline Files/Views/YourHome.xaml.cs:172]: This one is important. Currently we have a memory leak in Bundles because `IDisposable.Dispose()` in `YourHome.xaml.cs` is not called.

@yair100 fyi

Comment (d2dyno1) [reply Files/Views/YourHome.xaml.cs:172]: The solution is already in place but `YourHome.Dispose()` needs to be called which exceedes the scope of this PR :)

File: Files/ViewModels/Bundles/BundleItemViewModel.cs [hunk 4]
```diff
@@ -146,6 +172,18 @@ private async void SetIcon()
             {
                 try
                 {
+                    if (Path.EndsWith(".lnk"))
+                    {
+                        var (IconData, OverlayData, IsCustom) = await associatedInstance.FilesystemViewModel.LoadIconOverlayAsync(Path, 24u);
+
+                        await CoreApplication.MainView.ExecuteOnUIThreadAsync(async () =>
+                        {
+                            Icon = await IconData.ToBitmapAsync();
```
Comment (gave92) [inline Files/ViewModels/Bundles/BundleItemViewModel.cs:181]: IconData can be null

**Comment by gave92** [comment]
> Support for adding shourtcut files

How do you add shortcut files to a bundle since shortcuts can't be dragged?

**Comment by d2dyno1** [comment]
> > Option to rearrange Bundle items by dragging
> 
> Seems like I'm not able to drag items inside bundles, I can only rearrange bundles

It does work, but you must hold the item at the very edge of it. The reason it can't be dragged when directly clicking on it is because the button is clicked instead since it covers the entire gridviewrow area.

buttonInTheWay


@gave92 Do you happen to know a solution to such problem?

**Comment by gave92** [comment]
> @gave92 Do you happen to know a solution to such problem?

What about removing the Button and adding `IsItemClickEnabled/ItemClick` to the GridView?

**Comment by d2dyno1** [comment]
> > @gave92 Do you happen to know a solution to such problem?
> 
> What about removing the Button and adding `IsItemClickEnabled/ItemClick` to the GridView?

How would flyout work with it?
image

**Comment by gave92** [comment]
I'm positive `ContextFlyout` can be set on any element (e.g a Grid) not just Buttons

**Comment by d2dyno1** [comment]
@gave92 fixed! Thank you! 😊

File: Files/ViewModels/Bundles/BundleItemViewModel.cs [hunk 5]
```diff
@@ -146,6 +165,18 @@ private async void SetIcon()
             {
                 try
                 {
+                    if (Path.EndsWith(".lnk"))
+                    {
+                        var (IconData, OverlayData, IsCustom) = await associatedInstance.FilesystemViewModel.LoadIconOverlayAsync(Path, 24u);
+
+                        await CoreApplication.MainView.ExecuteOnUIThreadAsync(async () =>
+                        {
+                            Icon = await IconData?.ToBitmapAsync();
```
Comment (gave92) [inline Files/ViewModels/Bundles/BundleItemViewModel.cs:174]: Nope :) `await null` would still throw an exception.
```cs
if (IconData != null)`
{
    Icon = await IconData.ToBitmapAsync();
}```

Comment (gave92) [inline Files/ViewModels/Bundles/BundleItemViewModel.cs:174]: It's my bad sorry:
`await IconData?.ToBitmapAsync();` -> wrong, it will crash since you can't `await null`
`await IconData.ToBitmapAsync();` -> ok, since `ToBitmapAsync` checks for null

Comment (d2dyno1) [reply Files/ViewModels/Bundles/BundleItemViewModel.cs]: No worries 😅

**Review by gave92 — APPROVED** [review]
